src/coap_client.c

```diff
@@ -162,9 +177,24 @@ static enum golioth_status golioth_coap_client_set_internal(struct golioth_clien
 
     if (is_synchronous)
     {
-        // Created here, deleted by coap thread (or here if fail to enqueue
+        // Created here, deleted by coap thread (or here if fail to create or enqueue)
         request_msg.request_complete_event = golioth_event_group_create();
+        if (!request_msg.request_complete_event)
+        {
+            GLTH_LOGW(TAG, "Failed to create event group");
+            request_msg.status = NULL;
+            return GOLIOTH_ERR_MEM_ALLOC;
```

Comment:
We need to free `request_payload` allocated on line 160 above before returning.

Reply:
Good catch, thank you!

I made a subsequent commit to change how this is tested a few lines before this.